src/SqlAsyncCollector.cs

```diff
@@ -237,22 +252,57 @@ private async Task UpsertRowsAsync(IEnumerable<T> rows, SqlAttribute attribute,
         /// to upsert to and returns the extra property names in a List.
         /// </summary>
         /// <param name="columns"> The columns of the table to upsert to </param>
+        /// <param name="rows"> The rows to be upserted </param>
         /// <returns>List of property names that don't exist in the table</returns>
-        private static IEnumerable<string> GetExtraProperties(IDictionary<string, string> columns)
+        private static IEnumerable<string> GetExtraProperties(IDictionary<string, string> columns, IEnumerable<T> rows)
```

Comment:
I'd suggest just making this just take in a single item T - we only care about a single example to use for parsing so having the entire set isn't really necessary.